swiggyclone/service/views.py

```diff
@@ -22,45 +22,74 @@
-            total_price = 0
+            orderedBranch = data['branch']
             for orderDiscription in orderDiscriptions:
                 orderFoodItem = orderDiscription['foodItem']
                 orderQuantity = orderDiscription['quantity']
-                instance = FoodItem.filter(branch_id=orderedBranch, id=orderFoodItem)
-                foodItemPrice = instance.first().price
-                finalQuantity = instance.first().quantity - orderQuantity
-                serializer = FoodItemSerializer(instance.first(), data = {"quantity":finalQuantity}, partial = True)
-                if serializer.is_valid():
-                    serializer.save()
-                    total_price = total_price + orderQuantity*foodItemPrice
+                try: 
+                    instance = FoodItem.objects.filter(branch_id=orderedBranch, id=orderFoodItem)
+                except:
+                    orderValid = False
+                if orderValid:
+                    serialize = FoodItemSerializer(instance.first())
+                    foodItemQuantity = serialize.data["quantity"]
+                    if foodItemQuantity - orderQuantity<0:
+                        orderValid = False
+                          
+            if orderValid:
+                for orderDiscription in orderDiscriptions:
+                    orderFoodItem = orderDiscription['foodItem']
+                    orderQuantity = orderDiscription['quantity']
+                    instance = FoodItem.objects.filter(branch_id=orderedBranch, id=orderFoodItem)
+                    foodItemPrice = instance.first().price
+                    finalQuantity = instance.first().quantity - orderQuantity
+                    update_serializer = FoodItemSerializer(instance.first(), data = {"quantity":finalQuantity}, partial = True)
+                    if update_serializer.is_valid():
+                        update_serializer.save()
+                        total_price = total_price + orderQuantity*foodItemPrice
+                        
+                    else:
+                        return Response(update_serializer.errors,status = 400)
+                data["user"] = request.user
+                data["total_price"] = total_price
+                finalOrderSerializer = OrderSerializers(data=data)
+                if finalOrderSerializer.is_valid():   
+                    finalOrderSerializer.save()
                 else:
-                    return Response(serializer.errors,status = 400)
+                    orderValid = False
+                    return Response(orderSerializers.errors,status = 404)
         else:
-            Response(serializer.errors,status = 404)
+            orderValid = False
+            return Response(orderSerializers.errors,status = 404)
+        if orderValid:
+            return Response({"message":"Order Placed"}, status = 200)
+        return Response({"message":"Food Item is not present"},status = 404)
```

Comment:
if possible check those conditions in serializer and raise errors from serializer. raise ValidationError('...')